microsetta_public_api/api/emperor.py

```diff
@@ -1,44 +1,65 @@
 from microsetta_public_api.utils import jsonify
+from microsetta_public_api.utils._utils import stepwise_resource_getter
 from microsetta_public_api.repo._pcoa_repo import PCoARepo
 from microsetta_public_api.repo._metadata_repo import MetadataRepo
+from microsetta_public_api.resources_alt import get_resources
+from microsetta_public_api.config import schema
+from microsetta_public_api.api.metadata import _get_repo as _get_metadata_repo
+from microsetta_public_api.exceptions import UnknownResource
 
 
-def plot_pcoa_alt(beta_metric, named_sample_set, metadata_categories,
+def _get_pcoa_repo(dataset):
+    pcoas = stepwise_resource_getter(
+        get_resources(),
+        dataset,
+        schema.pcoa_kw,
+        'pcoa',
+    )
+    taxonomy_repo = PCoARepo(pcoas.data)
+    return taxonomy_repo
+
+
+def plot_pcoa_alt(dataset, beta_metric, named_sample_set, metadata_categories,
                   fillna='nan'):
-    raise NotImplementedError()
+    pcoa_repo = _get_pcoa_repo(dataset)
+    metadata_repo = _get_metadata_repo(get_resources)
+
+    return _plot_pcoa(beta_metric, fillna, metadata_categories,
+                      named_sample_set, metadata_repo, pcoa_repo)
 
 
 def plot_pcoa(beta_metric, named_sample_set, metadata_categories,
               fillna='nan'):
     pcoa_repo = PCoARepo()
     metadata_repo = MetadataRepo()
 
-    if not pcoa_repo.has_pcoa(named_sample_set, beta_metric):
-        return jsonify(
-                text=f"No PCoA for named_sample_set='{named_sample_set}',"
-                     f"beta_metric='{beta_metric}'",
-                code=404
-            ), 404
+    return _plot_pcoa(beta_metric, fillna, metadata_categories,
+                      named_sample_set, metadata_repo, pcoa_repo)
```

Comment:
Is there a reason not to merge these two functions?  Something like:

```
if dataset is None:
    pcoa_repo = PCoARepo()
    ...
else:
    pcoa_repo = get_pcoa_repo(dataset)
    ...
return _plot_pcoa(...)
```

Reply:
Plan is to deprecate the non-alt method and change the -alt method to main. Problem is we are adding a path parameter :) And connextion/flask won't let you have the same function for two different paths, iirc.